bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -604,9 +604,8 @@ public long calculateTableSize(TableId tableId, Optional<String> filter) {
 
   public long calculateTableSize(TableInfo tableInfo, Optional<String> filter) {
     TableDefinition.Type type = tableInfo.getDefinition().getType();
-    if (type == TableDefinition.Type.TABLE && !filter.isPresent()) {
```

Comment:
Is there a way to refresh `tableInfo`? `getNumRows()` is significantly faster

Reply:
Let me check.

Reply:
As discussed offline, let's go with this approach for now. Can be revisited later if optimization is required.